sources/packages/backend/apps/queue-consumers/src/processors/schedulers/ier-Integrations/ier-integration.scheduler.ts

```diff
@@ -0,0 +1,75 @@
+import { InjectQueue, OnQueueActive, Process, Processor } from "@nestjs/bull";
+import { IER12FileService } from "@sims/integrations/institution-integration/ier-integration";
+import {
+  IER_SCHEDULER_JOB_ID,
+  QUEUE_RETRY_DEFAULT_CONFIG,
+} from "@sims/services/constants";
+import { QueueNames } from "@sims/services/queue";
+import { PST_TIMEZONE } from "@sims/utilities";
+import { ConfigService } from "@sims/utilities/config";
+import { InjectLogger, LoggerService } from "@sims/utilities/logger";
+import Bull, { CronRepeatOptions, Job, Queue } from "bull";
+import { BaseScheduler } from "../base-scheduler";
+import {
+  GeneratedDateQueueInDTO,
+  IER12ResultQueueOutDTO,
+} from "./models/ier.model";
+
+@Processor(QueueNames.IERIntegration)
+export class IERIntegrationScheduler extends BaseScheduler<GeneratedDateQueueInDTO> {
+  cronOptions: Bull.JobOptions = undefined;
+
+  constructor(
+    @InjectQueue(QueueNames.IERIntegration)
+    readonly schedulerQueue: Queue<GeneratedDateQueueInDTO>,
+    private readonly ierRequest: IER12FileService,
+    config: ConfigService,
+  ) {
+    super();
+    // cron options.
+    this.cronOptions = {
+      ...QUEUE_RETRY_DEFAULT_CONFIG,
+      jobId: IER_SCHEDULER_JOB_ID,
```

Comment:
Is there any specific reason for passing the job id here?

Reply:
we can add multiple jobs to the same queue, for that we will need jobid, since here are going with separate queues, removing it